### PR TITLE
Documentation better release notes

### PR DESCRIPTION
**Pull request by biosafetylvl5**
* [ ] NO REQUIRED ***existing tests***
* [ ] NO REQUIRED ***unit tests***
* [ ] NO REQUIRED ***integration tests***
* [x] Required ***documentation*** added
* [ ] Required ***release notes*** added for new/modified functionality
* [ ] NO REQUIRED ***updates to other repos*** 

# Testing Instructions
Build the docs.

# Summary
I added top-level summaries to each version number. Eg. version X.0.0 -> X.10000.10000+ summarized under Version X. See changes for details 🙂 

# Output: Passes doc8

``` bash
➜  geoips git:(documentation-better-release-notes) doc8 --max-line-length 120 ./docs/source/releases/index.rst 
Scanning...
Validating...
========
Total files scanned = 1
Total files ignored = 0
Total accumulated errors = 0
Detailed error counts:
    - doc8.checks.CheckCarriageReturn = 0
    - doc8.checks.CheckIndentationNoTab = 0
    - doc8.checks.CheckMaxLineLength = 0
    - doc8.checks.CheckNewlineEndOfFile = 0
    - doc8.checks.CheckTrailingWhitespace = 0
    - doc8.checks.CheckValidity = 0
   ```

## Relevant XKCD

at least geoips isn't this bad

### CODE REVIEW

File: docs/source/releases/v1_14_0a0/documentation-better-release-notes.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+documentation:
+- description: 'Added summaries to the minor releases in the docs to make searching for a specific change easier.'
+  title: 'Add top level descriptions to minor releases'
+  files:
+    - modified: "docs/source/releases/index.rst"
```
Comment (mindyls) [inline docs/source/releases/v1_14_0a0/documentation-better-release-notes.yaml:5]: Will these work now?  Are we ready to start doing release notes this way?  @jsolbrig if so, we'll need to get the CI running, and update template_basic_plugin repo accordingly ?

File: docs/source/releases/index.rst [hunk 2]
```diff
@@ -175,6 +220,11 @@ Version 1.2
 Version 1.1
 -----------
 
+Addition of new features like MODIS reader, stitched product output, sector overpass database, and Visible global
+stitched. Refactoring for Python 3 operation, and resolving errors in MODIS processing and RGB MTIF color issue. There
+are also breaking changes including modularized PMW algorithms, changed dictionary structures, and standardized platform
+names.
+
```
Comment (mindyls) [inline docs/source/releases/index.rst:227]: Oh - what do wthese look like when the documentation is built?

**Comment by biosafetylvl5** [comment]
Screenshot 2024-07-15 at 2 48 40 PM

^ what it looks like

File: docs/source/releases/index.rst (hunk 2 above)
Comment (mindyls) [reply docs/source/releases/index.rst:227]: Thanks!

File: docs/source/releases/v1_14_0a0/documentation-better-release-notes.yaml (hunk 1 above)
Comment (mindyls) [reply docs/source/releases/v1_14_0a0/documentation-better-release-notes.yaml:5]: Pending @jsolbrig determination if we are ready to get this CI set up right away

**Review by mindyls — APPROVED** [review]
Thanks for this!  It is interesting reading back through the old summaries

**Review by jsolbrig — APPROVED** [review]

**Pull request merged**